api/src/metrics/ProxyMeter.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Meter, MeterOptions } from './Meter';
+import { NoopMeter } from './NoopMeter';
+import {
+  BatchObservableCallback,
+  Counter,
+  Histogram,
+  MetricOptions,
+  ObservableCounter,
+  ObservableGauge,
+  ObservableUpDownCounter,
+  UpDownCounter,
+  Observable,
+  Gauge,
+} from './Metric';
+
+const NOOP_METER = new NoopMeter();
+
+/**
+ * Proxy meter provided by the proxy meter provider
+ */
+export class ProxyMeter implements Meter {
+  // When a real implementation is provided, this will be it
+  private _delegate?: Meter;
+
+  constructor(
+    private _provider: MeterDelegator,
+    public readonly name: string,
+    public readonly version?: string,
+    public readonly options?: MeterOptions
+  ) {}
+
+  /**
+   * @see {@link Meter.createGauge}
+   */
+  createGauge(_name: string, _options?: MetricOptions): Gauge {
+    return this._getMeter().createGauge(_name, _options);
+  }
```

Comment:
There's a certain difficulty in the case of the metrics API vs the trace/logs APIs - there's another layer that needs to be taken into account, which is the instruments. 

In Traces, the (no-op) span is an object which is only kept for the time that the span is active. So one span being no-op is not too problematic. For logs, the record is passed straight to the logger so it's also not a problem there. In metrics however, an instrument is instantiated early and is kept over the lifetime of the app. That means that when a user gets a Meter, they usually also create the Instruments already. A later-updated `ProxyMeter` would therefore be of very limited help as the instruments would also need to be proxies in order to give a smooth experience.

Reply:
Thanks for taking a look @pichlermarc, I wanted to fix #4112, so this is still relevant for us, I understand this change may need a major version update in the api package, maybe this is something I can move to a different branch and start experimenting with Instruments proxies or similar concept, what do you think?

Reply:
I don't think the major bump is necessary for this (just as with logs and traces). The types we export from the API are what we guarantee will be there. If there's extra properties on the registered SDK and those are available on the API too in plain JavaScript, then that was never guaranteed to also be there when returned by the API.

We can merge this to a separate branch to experiment with delegating instruments, but I think that could also be done here on this branch, as the PR won't really address the whole problem (instruments created before SDK registration being no-op) as it stands right now.